### ISSUE TITLE
Recording unique call pairs

### ISSUE THREAD

**Issue by sh1r1sh** [namhyung/uftrace#813]
Hi Namhyung,
I wanted to know if it will be possible to add an enhancement that:
1. Captures only the unique call pairs (caller-callee). i.e. no new entries will be recorded after the first time.
2. Has the ability to support a reset/clear so that the unique call pairs will be saved after the reset/clear.

I have a daemon program and I would like to derive the runtime call chain at different points in time. Ex:
Step 1. Start program recording unique call pairs
Step 2. Store the recorded file
Step 3. Reset/clear the recording mechanism
Step 3. Send new input to the daemon program
Step 4: Store the recorded file
Step 5: Reset the recording mechanism
Step 6. Send new input
and so on.

**Comment by honggyukim** [comment]
Hi @sh1r1sh, thanks for using uftrace.

uftrace doesn't have builtin support for that, but I think you can write a python script to make the unique call pair. uftrace can provide a hook for each function entry and exit and you can write `uftrace_entry` and `uftrace_exit` python function, then do what you want inside them.

Please read the python scripting section in the following tutorial slide.
- https://uftrace.github.io/slide/#python

I think you better first record the trace then use `script` command for post processing with your script.

**Comment by namhyung** [comment]
@sh1r1sh Hello,thanks for the suggestion.  Is it to reduce the size of tracing data or just to make analysis easier?

**Comment by sh1r1sh** [comment]
> @sh1r1sh Hello,thanks for the suggestion. Is it to reduce the size of tracing data or just to make analysis easier?

Yes, it is to reduce the size of the tracing data as we have a long running daemon process and for my use, the unique caller-callee pair is sufficient. I will experiment a few options and see what works.

**Comment by namhyung** [comment]
Sorry for the late reply.  I think it's a good idea to reduce repeated data from record but it somehow needs to keep the information.  And I don't want to make the fast path slow because of this, will think about it more later.
